cmake/modules/MOLD.cmake

```diff
@@ -0,0 +1,43 @@
+option(ENABLE_MOLD "Use MOLD linker" OFF)
+if(ENABLE_MOLD AND NOT MOLD_FOUND)
+  if(ENABLE_LDGOLD OR ENABLE_LLD)
+    message(FATAL_ERROR "Only enable ENABLE_MOLD OR ENABLE_LDGOLD OR ENABLE_LLD")
+  endif()
+
+  if(CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 12)
+    # GCC < 12 doesn't support -fuse-ld=mold, so we have to use tools prefix path
+    # if mold is installed in a non-standard dir, users can set -DMOLD_PREFIX=/path/to/mold_install_prefix
+    find_path(MOLD_PREFIX_DIR NAMES ld
+                              NO_DEFAULT_PATH # this is needed, otherwise we find /usr/bin/ld
+                              PATH_SUFFIXES libexec/mold
+                              PATHS ${MOLD_PREFIX} /usr /usr/local)
+    if(MOLD_PREFIX_DIR)
+      set(COMPILER_ARGS "-B${MOLD_PREFIX_DIR}")
+    else()
+      message(WARN "Mold LD path not found, you might need to set -DMOLD_PREFIX=/path/to/mold_install_prefix")
```

Comment:
is `WARN` a valid level? The cmake docs say `WARNING`. https://cmake.org/cmake/help/latest/command/message.html